### DISCUSSION TITLE
Make V's Install Less Troublesome

### DISCUSSION THREAD

**Discussion by Wajinn** [vlang/v#13747] [General]
I recommended V to a friend, who I then had to help.  The installation instructions could be better.  At least a bit more verbose and reflect what users will actually need to do and could encounter.  Possibly a couple of paragraphs, on install troubleshooting.  Of course every situation can not be known beforehand, but some tips and suggestions could go a long way.

Otherwise, V could unnecessarily frustrate or lose a number of potential users who may go somewhere else, where things are more organized.  Yes, V is not 1.0 yet, but hopefully such a great language will be getting more attention.  Plus, first impressions are important. 

If we go to https://github.com/vlang/v, there is the following.

```
git clone https://github.com/vlang/v
cd v
.\make.bat
```
1) Various potential users of the language will not be Git users.

Maybe the argument might be that V is for "developers only", but that would be both unlikely and or a bit shortsighted, as enthusiastic novices to advanced users may eventually go on to become contributors or developers.

2) If attempting Git and clone, Windows users can get errors.  

A) Errors because v_win.c doesn't exits (can be downloaded from https://github.com/vlang/vc)
B) Errors because the TCC folder doesn't exist or can't be downloaded
C) Errors related to Git, because it's not set in the Path or cmd\git.exe can't be found

3) If have managed to get set up, then there can be issues with V's "install" at cmd not always working.

This can be because the metadata for the module they may want doesn't exits.  It would be good to show where and how to manually add modules (`\vlib `or `\.vmodules`).

4) Maybe some extra info on Git downloads.

**Comment by ylluminate** [comment]
A thought that has remained at the front of my mind in this regard since the beginning in 2019 has been: the lowest entry point for any language, and arguably what spring-boarded JavaScript, is in-browser accessibility. This allows someone the ability to just "jump in" and start playing without thinking about any kind of setup. To that end I believe the WASM effort is pinnacle and having a robust sandbox/playground within the browser with this kind of experience. jsbin.com, jsfiddle.net, codepen.io, and others are interesting possibilities for models.

Obviously the current and best path is just installing it on your local machine and that which you say can be problematic. It's nice that you've attempted to outline your friend's experience somewhat. I guess many of us have our heads buried in other things right now and/or work in operating systems that generally have the precursory tooling in place. 

Perhaps you can work with your friend and some other fresh V inductees in coming up with a solution that you think would be foolproof in the majority of circumstances? It's hard to cover all of the edge cases, of course, and it seems like maybe your friend hit one - BUT maybe not, maybe it's just that there is a better way that either we haven't seen yet or perhaps haven't had time to implement yet... 

So yeah, I definitely think your introductory share above is great, especially for you or someone with similar motivation/expectations to work through more details or suggestions of what you feel would give that desired "BAM!" first experience... I'm sure there are at least a few folks around who would be happy to take a cohesive plan and implement it.

**Reply by Wajinn** [reply]
In the context that the playground is something like a structured online web tutorial and training, that would be great, but appears it would be an extensive project by itself.

I disagree about the browser being the best point of entry for everyone, because there can be privacy related issues.  Various people may want to learn privately or don't want anything that they create exposed on the internet.  It's hard to say what percentage of people would have privacy concerns or are not yet doing anything where it would be an issue.

V to WASM is awesomeness by itself.

Yes, I do think that the documentation could be augmented.  Maybe an extra paragraph.  Could put the suggested paragraph in issues.  I'm thinking that letting people know how to handle some potential "gotchas", would be helpful for people new to V.

**Reply by ylluminate** [reply]
No, I mean a more freeform tool that allows one to just start plugging code in and playing as per the examples. 

Depends on the kinds of projects you're talking about. I think for maximum acceptance for students and introductory use, it is ideal. For those with more experience and/or commitment to go further, yeah, a shell installation is clearly the best route.

**Comment by changrui** [comment]
Download the excutable zip here: `https://github.com/vlang/v/tags` depend on your OS please.

**Reply by Wajinn** [reply]
Yes, there are zips, but that wasn't exactly my point.  If the person downloads the zip, that won't resolve Git issues, update or upgrade issues, nor show what to do if the metadata for a module isn't there.  So, thinking the documentation could have an extra paragraph or so on potential or common issues, that new users may encounter.

**Reply by ylluminate** [reply]
If you are indeed talking about documentation, please feel free to make a PR based on the experiences you had via your friend. That'd be great and perhaps that will be the end of this discussion if you feel that that is what is needed @Wajinn.

**Comment by ylluminate** [comment]
@Wajinn please see: https://github.com/vlang/v/commit/d87b5d40b66280409527c8f2c9900dc655f66f66

Notice the new Discussions section linked to: https://github.com/vlang/v/discussions/categories/installation-issues

I've created a Windows entry where you're welcome to please share details that helped you / your friend: https://github.com/vlang/v/discussions/13768

Do you feel that this approach for inviting and putting forth of more information is helpful?

For what it's worth, I've seen folks mention winlibs.com as being very effective for V Windows-based installations since it's a "drop in" style solution. It can also be easily/cleanly removed as well from what I've heard. Please search the Discord discussions for more details and caveats. It also appears @Nukiloco has had some experience here: https://github.com/vlang/v/issues/11951#issuecomment-925868080

Again, it would be nice if someone with Windows V experience could flesh out some helpful information in the new [Windows installation discussion](https://github.com/vlang/v/discussions/13768).
